common-solidity/coordination-contract/contracts/CrosschainCoordinationV1.sol

```diff
@@ -473,11 +508,148 @@ contract CrosschainCoordinationV1 is CrosschainCoordinationInterface {
-        bytes memory pubKey = sidechains[_sidechainId].publicKey;
-        return pubKey;
+ //   function getPublicKey(uint256 _sidechainId) external view returns (bytes memory) {
+ //       bytes memory pubKey = sidechains[_sidechainId].publicKey;
+ //       return pubKey;
+ //   }
+
+
+    /**
+     * Get blockchain's public key, version number, status and block number
+     *
+     * @param _sidechainId The 256 bit sidechain identifier to which this public key belongs
+     * @return an array of public keys for the sidechain corresponding to the 3 different states it can be in:
+     *         Value      Status
+     *           0        This is the active public key for the sidechain which is currently in use
+     *           1        The public key that has been returned is flagged as a proposed changed key, so it is dependent on the voting before it can become active
+     *           2        This is a public key that has been used previously, but is currently not in use. Note that this key could be the prior key, or an historic key
+     */
+    function getPublicKey(uint256 _sidechainId) external view returns ( uint _versionNumber, uint256 _status, uint _blockNumber, bytes memory _key){
+
+        (uint256 vsn, uint256 stat, uint256 blk, bytes memory key) = getLastActivePublicKey(_sidechainId);
+
+        return (vsn, stat, blk, key);
     }
 
+
+    /**
+     * Get array of Sidechain's public key, version number, status and block number
+     *
+     * @param _sidechainId The 256 bit sidechain identifier to which this public key belongs
+     * @return an array of public keys for the sidechain corresponding to the 3 different states it can be in:
+     *         Value      Status
+     *           0        This is the active public key for the sidechain which is currently in use
+     *           1        The public key that has been returned is flagged as a proposed changed key, so it is dependent on the voting before it can become active
+     *           2        This is a public key that has been used previously, but is currently not in use. Note that this key could be the prior key, or an historic key
+     */
+    function getPublicKeyArray (uint256 _sidechainId) private view returns ( PublicKey [] memory _publicKeyArray){
+        uint256 sidechainsNoOfKeys = sidechains[_sidechainId].numberOfKeys;
+        PublicKey [] memory publicKeyArray;
+        for (uint i = 0; i < sidechainsNoOfKeys; i++) {
+            publicKeyArray[i] = sidechains[_sidechainId].publicKeys[i];
+        }
+        return(publicKeyArray);
+    }
+
+    /**
+     * Get the last active key for the sidechain
+     *
+     * @param _sidechainId The 256 bit sidechain identifier to which this public key belongs
+     * @return _activeKey latest active public key
+     */
+    function getLastActivePublicKey(uint256 _sidechainId) private view returns ( uint _versionNumber, uint256 _status, uint _blockNumber, bytes memory _key){
```

Comment:
If only keys that are successfully voted for are stored in the sidechains[_sidechainId].publicKeys data structure, then the latest public key information will be available at: sidechains[_sidechainId].publicKeys[sidechains[_sidechainId].numberOfKeys]

Reply:
We need to talk about this